### Write a Python function, find_square() that accepts an integer number n and returns the square of n. Invoke the function and display the square of the number. 

In [1]:
def find_square(n):
    square = n ** 2
    return square

In [2]:
find_square(15)

225

### Write a Python function, find_sum() that accepts an integer n and returns the sum of first n numbers. Invoke the function and display the sum of first n numbers.

In [3]:
def sum_n_numbers(n):
    s = 0
    for i in range(1,n+1):
        s += i
        
    return s

In [4]:
sum_n_numbers(100)

5050

### Write a Python function factorial(num) which returns the factorial of a given number.

In [5]:
def Factorial(n):
    fact = 1
    for i in range(1,n+1):
        fact *= i
    return fact

In [6]:
Factorial(5)

120

### Write a Python Function is_palindrome(num) that accepts an integer num as argument and returns True if the num is palindrome else returns false. Invoke the function and based on return value print the output.

Example: num=12321 output: Given number is a palindrome, num=12345 output: Given number is not a palindrome



In [7]:
def CheckPalindrome(n):
    str_n = str(n)
    if str_n == str_n[::-1]:
        print("It is a palindrome number")
    else:
        print("It is not a palindrom number")

In [8]:
CheckPalindrome(12321)

It is a palindrome number


In [9]:
CheckPalindrome(321232)

It is not a palindrom number


### Write a Python function check_amicable_numbers(num1, num2) that accepts two numbers num1 and num2 as arguments and returns True if the given pair of numbers are amicable numbers else return false. Invoke the function and based on return value print the numbers are amicable numbers or not.

num1 and num2 are said to be amicable numbers if sum of all the proper devisors (except num1 itself) of num1 is equal to num2 and sum of all the proper devisors of num2 (except num1 itself) is equal to num1.

Example: 220 and 284 are amicable numbers as

Proper devisors of 220 are 1, 2, 4, 5, 10, 11, 20, 22, 44, 55, 110 whose sum is 284

Proper devisors of 284 are 1, 2, 4, 71, 142 whose sum is 220

In [16]:
def Get_Factorials(n):
    '''
    Gets the given numbers devisors, excluding given number
    '''
    facts = []
    for i in range(1,n):
        if n%i == 0:
            facts.append(i)
    return facts

In [13]:
def CheckAmicable(num1,num2):
    num1_facts = Get_Factorials(num1)
    num2_facts = Get_Factorials(num2)
    
    if sum(num1_facts) == num2 and sum(num2_facts) == num1:
        print("It is a Amicable number")
    else:
        print("It is not a Amicable number")

In [14]:
CheckAmicable(220,284)

It is a Amicable number


In [15]:
CheckAmicable(202,220)

It is not a Amicable number


### Write a Python function right_shift(num, n) that takes two numbers num and n as  arguments and returns value of the integer num rotated to the right by n positions. Assume both the numbers are unsigned. Invoke the function and print the return value.

Hint: use >> binary operator to shift the bits.

Example: num=60, n=2 result=15

In [18]:
def right_shift(num,n):
    return num>>n

In [19]:
right_shift(60,2)

15

In [20]:
right_shift(50,3)

6

### Write a Python function check_strong_number(num) that accepts a positive integer as argument and returns True if the number is strong number else false. Invoke the function and based on return value print the number is strong number or not.

A number is said to be strong number, if the sum of factorial of each digit of the number is equal to the given number.

Example:145 is strong number as

1!=1

4!=24

5!=120

Sum of all these is 145

In [23]:
def CheckStrongNumber(num):
    str_num = str(num)
    import math
    s = 0
    for i in str_num:
        s += math.factorial(int(i))
            
    if s == num :
        print("It is a strong number")
    else:
        print("It is not a strong number")

In [24]:
CheckStrongNumber(145)

It is a strong number


In [25]:
CheckStrongNumber(156)

It is not a strong number
